TC-44 personal protection equipment classification list - test case automation

wcolina opened the pull request:
Automated test case for "TC-44 personal protection equipment classification list".

wcolina committed:
Added automated test case for:
TC-44 Clasificacion de equipo de proteccion personal - Todos los datos de clasificacion de equipos de proteccion personal son mostrados en una lista.

wcolina committed:
Added automated test case for:
                                                                                                                                      TC-44 Clasificacion de equipo de proteccion personal - Todos los datos de clasificacion de equipos de proteccion personal son mostrados en una lista.

Refactor code with more appropriate variable and method names.

wcolina commented:
Test case:
![image](https://user-images.githubusercontent.com/852076/41744876-211153ac-7573-11e8-9739-8a5a9b2f7384.png)

Automated test case result:
![image](https://user-images.githubusercontent.com/852076/41744893-30b16c5c-7573-11e8-8818-032179c45b55.png)

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -17,16 +18,26 @@
     @FindBy(id = "personnelMenu")
     private WebElement personnelMenu;
 
+    @FindBy(id = "ppeMenu")
```
please format your code
user abbreviations

wcolina replied:
Done.

wcolina replied:
Fixed observations.

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -17,16 +18,26 @@
+    @FindBy(id = "ppeClassificationSubMenu")
+    private WebElement ppeClassificationSubMenu;
+
     public void clickOnPersonnelMenu(){
```
user clickPersonnelMenu instead of clickOnPersonnelMenu, for all method in this class

marcelogaray commented on src/test/resources/features/personalProtectionEquipment/PpeClassificationList.feature:
```diff
@@ -0,0 +1,8 @@
+  Scenario: Clasificacion de equipo de proteccion personal - Todos los datos de clasificacion de equipos de proteccion personal son mostrados en una lista
+    Given 'SSI-A' home page is loaded
+    And Clicking on PPE menu on 'Header' page
+    And Click in sub menu 'PPE Classification' of menu 'PPE'
```
click 'PPE Classification' menu on 'PPE' page

marcelogaray commented on src/test/resources/features/personalProtectionEquipment/PpeClassificationList.feature:
```diff
@@ -0,0 +1,8 @@
+Feature: PpeClassificationList
+  Scenario: Clasificacion de equipo de proteccion personal - Todos los datos de clasificacion de equipos de proteccion personal son mostrados en una lista
+    Given 'SSI-A' home page is loaded
+    And Clicking on PPE menu on 'Header' page
```
And click 'PPE' menu on 'Header' page

marcelogaray reviewed (changes requested):
review observations

wcolina committed:
Format code.
Improved imports.
Using abbreviation PPE for personal protection equipment in the package.
Using PPE acronym in method's name.
Refactor method names following correct format, deleted word 'On' and changed 'Click' instead of 'Clicking' in the steps/feature definitions.

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -1,11 +1,12 @@
-import org.umssdiplo.automationv01.core.managepage.employee.EmployeeList;
-import org.umssdiplo.automationv01.core.managepage.role.RoleList;
-import org.umssdiplo.automationv01.core.utils.CommonEvents;
+        import org.openqa.selenium.WebElement;
```
you added spaces. please update this one.

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -17,17 +18,27 @@
+        CommonEvents.clickButton(ppeMenu);
+    }
+
+    public EmployeeList clickOnEmployeeMenu() {
```
please change the name to 'clickEmployeeMenu'

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java:
```diff
@@ -0,0 +1,19 @@
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class PPEClassificationList extends BasePage {
```
this page should have a constructure where it should wait for something in specia.

marcelogaray reviewed (changes requested):
review observations

marcelogaray commented:
Please fix conflicts and my observations
Send me a pull request once you fix it, I will close it in the meantime
